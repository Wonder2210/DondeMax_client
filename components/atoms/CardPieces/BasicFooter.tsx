import React from "react";
import { Flex, FlexProps } from "@chakra-ui/react";

interface props extends FlexProps {
  borders?: [string, string];
}

const BasicFooter: React.FC<props> = ({
  children,
  position = "static",
  zIndex,
  width = "100%",
  height = "0",
  right = "0",
  bottom = "0",
  left = "0",
  top = "0",
  backgroundColor = "#FFF",
  borders = ["35px", "35px"],
}) => {
  return (
    <Flex
      data-testid="basicfooter"
      backgroundColor={backgroundColor}
      zIndex={zIndex ?? 0}
      pos={position}
      top={top}
      right={right}
      bottom={bottom}
      left={left}
      width={width}
      height={height ?? ""}
      align="center"
      textAlign="center"
      justify="center"
      roundedTop={borders[0]}
      roundedBottom={borders[1]}
    >
      {children}
    </Flex>
  );
};

export default BasicFooter;
