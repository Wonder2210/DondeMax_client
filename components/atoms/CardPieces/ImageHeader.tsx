import * as React from "react";
import { Image } from "@chakra-ui/core";

type props = {
  src: string;
  alt: string;
  height?: string;
  width?: string;
};

const ImageHeader: React.FC<props> = ({ src, alt, height, width }) => {
  return <Image src={src} alt={alt} height={height ?? ""} width={width ?? ""} rounded="35px" />;
};

export default ImageHeader;
