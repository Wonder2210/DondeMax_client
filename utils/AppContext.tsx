import * as React from "react";
import { useDisclosure } from "@chakra-ui/react";
import Cookies from "js-cookie";

export const AppContext = React.createContext(null);

type ProductsCart = {
  id: number;
  name: string;
  price: number;
  image: string;
  quantity: number;
  total: number;
};

export function useAppContext() {
  const context = React.useContext(AppContext);

  if (!context) {
    // eslint-disable-next-line no-console
    console.error("Error deploying App Context!!!");
  }

  return context;
}

export const AppProvider: React.FC = ({ children }) => {
  const [state, setState] = React.useState<{
    admin: boolean;
    adminPassword: number;
    productsCart?: Array<ProductsCart>;
    total: number;
    authToken: string;
    cart: boolean;
  }>(() => {
    const admin = Cookies.get("admin") === "true" ?? false;
    const adminPassword = 221099;
    const productsCart = Cookies.getJSON("productsCart") ?? [];
    const authToken = Cookies.get("authToken") ?? "";

    return {
      admin,
      adminPassword,
      productsCart,
      total: 0,
      authToken,
      cart: false,
    };
  });
  React.useEffect(() => {
    if (state.productsCart.length) {
      setState({ ...state, total: state.productsCart.reduce((prev, current) => prev + current.total, 0) });
    }
  }, [state.productsCart]);

  const addToCart = (product: ProductsCart) => {
    const itemIndex = state.productsCart.findIndex((i) => i.id === product.id);

    if (itemIndex >= 0) {
      const items = [...state.productsCart];
      items[itemIndex].quantity += product.quantity;
      items[itemIndex].total += product.total;

      setState({ ...state, productsCart: items });
      return;
    }
    setState({ ...state, productsCart: [...state.productsCart, product] });
  };

  const removeFromCart = (id: number) => {
    setState({ ...state, productsCart: state.productsCart.filter((i) => i.id !== id) });
  };

  const openCart = () => setState({ ...state, cart: true });
  const closeCart = () => setState({ ...state, cart: false });

  const setAuthToken = (e: string) => setState({ ...state, authToken: `Bearer ${e}` });
  React.useEffect(() => {
    const stateJ = JSON.stringify(state.productsCart);
    Cookies.set("admin", String(state.admin), { expires: 1 });
    Cookies.set("adminPassword", String(state.adminPassword), { expires: 1 });
    Cookies.set("productsCart", stateJ, { expires: 1 });
    Cookies.set("authToken", state.authToken);
  }, [state]);

  const values = React.useMemo(
    () => ({
      state,
      setState,
      addToCart,
      removeFromCart,
      setAuthToken,
      openCart,
      closeCart,
    }),
    [state],
  );

  return <AppContext.Provider value={values}>{children}</AppContext.Provider>;
};
