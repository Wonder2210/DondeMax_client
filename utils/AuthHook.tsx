import * as React from "react";
import { useQuery, gql } from "@apollo/client";
import Cookie from "js-cookie";

const query = gql`
  query SessionUser {
    sessionUser {
      id
      phone
      name
      role
      email
    }
  }
`;

export const useAuth = () => {
  const [user, setUser] = React.useState({
    id: 0,
    name: "",
    role: "",
    email: "",
  });

  const { data, loading, error, refetch } = useQuery(query);

  React.useEffect(() => {
    if (!loading && !error) {
      setUser({
        id: data.sessionUser.id,
        name: data.sessionUser.name,
        role: data.sessionUser.role,
        email: data.sessionUser.email,
      });
    }
  }, [data]);

  return {
    user,
    refetch,
  };
};
