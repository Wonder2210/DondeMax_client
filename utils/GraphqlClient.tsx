import * as React from "react";
import Cookies from "js-cookie";
import { ApolloClient, InMemoryCache, ApolloProvider } from "@apollo/client";
import { setContext } from "@apollo/client/link/context";
import { createUploadLink } from "apollo-upload-client";
import fetch from "cross-fetch";

const Client = (token) => {
  const setAuthorizationLink = setContext((request, previousContext) => ({
    headers: { authorization: token },
  }));

  const link = new createUploadLink({ uri: "http://localhost:4000/graphql", fetch });
  return new ApolloClient({
    link: setAuthorizationLink.concat(link),
    cache: new InMemoryCache(),
  });
};

const GraphqlProvider: React.FC = ({ children }) => {
  return <ApolloProvider client={Client(Cookies.get("auth"))}>{children}</ApolloProvider>;
};

export default GraphqlProvider;
