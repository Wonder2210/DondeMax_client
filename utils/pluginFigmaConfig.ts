export default {
    name: 'Button',
    parameters: {
      design: {
        type: 'figma',
        url: 'https://www.figma.com/file/5OkUaPf8h9zZQEO7H04Odk/DondeMax-Web',
      },
    },
  }