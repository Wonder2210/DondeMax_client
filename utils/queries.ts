import { gql } from "@apollo/client";

export const GET_TYPES = gql`
  query types {
    productTypes {
      id
      type
    }
    productPreservation {
      id
      type
    }
  }
`;

export const GET_DATA = gql`
  query ProductsData($size: Int! = 12, $cursor: Int! = 0, $type: String = "", $preservation: String = "") {
    products(size: $size, cursor: $cursor, type: $type, preservation: $preservation) {
      total
      results {
        id
        name
        image
      }
    }
  }
`;

export const GET_BASE_PRODUCTS = gql`
  query GetProductsBase($size: Int! = 12, $cursor: Int! = 0, $type: String = "", $preservation: String = "") {
    products(size:$size, cursor:$cursor, type:$type, preservation:$preservation ){
    total
    results{
      id
      name
      image
      info
      type
    }
  }
  }
`;


export const GET_PROVIDERS = gql`
  query GetProviders {
    providers {
      id
      name
      RIF
      phone
      direction
    }
  }
`;

export const CREATE_PROVIDER = gql`
  mutation CreateProvider($name: String!, $RIF: String!, $phone: String!, $direction: String!) {
    createProvider(provider: { name: $name, RIF: $RIF, phone: $phone, direction: $direction }) {
      id
    }
  }
`;

export const UPDATE_PROVIDER = gql`
  mutation UpdateProvider($id: Int!, $name: String, $RIF: String, $phone: String, $direction: String) {
    updateProvider(provider: { id: $id, name: $name, RIF: $RIF, phone: $phone, direction: $direction }) {
      id
    }
  }
`;

export const DELETE_PROVIDER = gql`
  mutation DeleteProvider($id : Int!){
  deleteProvider(id:$id)
}
`;

export const GET_MERCANCIA = gql`
  query GET_MATERIALS {
    materials {
      id
      nombre
      onStock{
        uniteds
        weight
      }
      type {
        id
        name
      }
    }
    materialTypes {
      id
      type: name
    }
  }
`;

export const CREATE_MATERIAL = gql`
  mutation CreateMaterial($nombre: String!, $type: Int!) {
    createMaterial(material: { nombre: $nombre, type: $type }) {
      id
    }
  }
`;

export const UPDATE_MATERIAL = gql`
  mutation UpdateMaterial($id: Int!, $nombre: String!, $type: Int!) {
    updateMaterial(id: $id, material: { nombre: $nombre, type: $type }) {
      id
      nombre
    }
  }
`;

export const DELETE_MATERIAL = gql`
  mutation DeleteMaterial($id: Int!) {
    deleteMaterial(id: $id)
  }
`;
