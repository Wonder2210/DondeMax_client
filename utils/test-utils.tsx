import * as React from "react";
import { render, RenderResult } from "@testing-library/react";
import { ChakraProvider } from "@chakra-ui/react";
import theme from "../utils/theme";

const Wrappers = ({ children }) => {
  return <ChakraProvider theme={theme}>{children}</ChakraProvider>;
};

const customRender = (ui: React.ReactElement, options?): RenderResult => {
  return render(ui, { wrapper: Wrappers, ...options });
};

export { customRender as render };
