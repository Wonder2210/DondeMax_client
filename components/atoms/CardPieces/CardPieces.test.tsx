import * as React from "react";

import { render } from "../../../utils/test-utils";
import BasicFooter from "./BasicFooter";
import ImageHeader from "./ImageHeader";

describe("Card pieces", () => {
  test("basic footer", () => {
    const { asFragment } = render(<BasicFooter>here i am</BasicFooter>);

    expect(asFragment()).toMatchSnapshot();
  });

  test("ImageHeader", () => {
    const { asFragment } = render(<ImageHeader alt="placeholder" src="https://via.placeholder.com/150" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
