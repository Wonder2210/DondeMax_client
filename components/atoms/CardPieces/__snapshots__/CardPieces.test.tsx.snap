// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card pieces ImageHeader 1`] = `
<DocumentFragment>
  <img
    alt="placeholder"
    class="chakra-image__placeholder css-qa3p9c"
  />
</DocumentFragment>
`;

exports[`Card pieces basic footer 1`] = `
<DocumentFragment>
  <div
    class="css-qynvi5"
    data-testid="basicfooter"
  >
    here i am
  </div>
</DocumentFragment>
`;
